<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Section 7: MULTIVARIATE AND NETWORK ANALYSES • diffTop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Section 7: MULTIVARIATE AND NETWORK ANALYSES">
<meta property="og:description" content="diffTop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">diffTop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/diffTop.html">Get started</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/appendix/01_introduction.html">Section 1: INTRODUCTION</a>
    </li>
    <li>
      <a href="../../articles/appendix/02_microbiome_data.html">Section 2: MICROBIOME DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/03_example_16S_data.html">Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/04_models.html">Section 4: MODELS</a>
    </li>
    <li>
      <a href="../../articles/appendix/05_transformation.html">Section 5: TRANSFORMATIONS</a>
    </li>
    <li>
      <a href="../../articles/appendix/06_visualization.html">Section 6: VISUALIZATIONS FOR HETEROGENEOUS DATA</a>
    </li>
    <li>
      <a href="../../articles/appendix/07_multivariate_and_network_analysis.html">Section 7: MULTIVARIATE AND NETWORK ANALYSES</a>
    </li>
    <li>
      <a href="../../articles/appendix/08_differential_abundance_analysis.html">Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</a>
    </li>
    <li>
      <a href="../../articles/appendix/09_latent_Dirichlet_allocation.html">Section 9: LATENT DIRICHLET ALLOCATION</a>
    </li>
    <li>
      <a href="../../articles/appendix/10_uncertainity_quan_ordination.html">Section 10: UNCERTAINTY QUANTIFICATION FOR ORDINATION ANALYSIS</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PratheepaJ/diffTop/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="07_multivariate_and_network_analysis_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Section 7: MULTIVARIATE AND NETWORK ANALYSES</h1>
                        <h4 class="author">Pratheepa Jeganathan</h4>
            
            <h4 class="date">09 July, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PratheepaJ/diffTop/blob/master/vignettes/appendix/07_multivariate_and_network_analysis.Rmd"><code>vignettes/appendix/07_multivariate_and_network_analysis.Rmd</code></a></small>
      <div class="hidden name"><code>07_multivariate_and_network_analysis.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">genefilter</span><span class="op">)</span> <span class="co">#KOverA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span> <span class="co"># geom_text_repel</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">randomcoloR</span><span class="op">)</span><span class="co"># distinctColorPalette(n)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">adaptiveGPCA</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org">igraph</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jfukuyama/phyloseqGraphTest">phyloseqGraphTest</a></span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/load_all.html">load_all</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">theme_set</span><span class="op">(</span><span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu">theme_update</span><span class="op">(</span>
  text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"psE_BARBI"</span><span class="op">)</span>
<span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genefilter/man/kOverA.html">kOverA</a></span><span class="op">(</span><span class="fl">2</span>, A <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> 
<span class="va">psE_BARBI</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/filter_taxa.html">filter_taxa</a></span><span class="op">(</span>
  <span class="va">psE_BARBI</span>, 
  <span class="va">threshold</span>, <span class="cn">TRUE</span><span class="op">)</span> 

<span class="va">psE_BARBI</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">psE_BARBI</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">psE_BARBI</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">ps</span><span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1418 taxa and 86 samples ]
## sample_data() Sample Data:       [ 86 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1418 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1418 tips and 1417 internal nodes ]</code></pre>
<div id="edit-specimen-names" class="section level3">
<h3 class="hasAnchor">
<a href="#edit-specimen-names" class="anchor"></a>Edit specimen names</h3>
<p>We edit specimen names and identify Asteraceae and non-Asteraceae plants.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, <span class="st">"E106"</span>, <span class="st">"E-106"</span><span class="op">)</span>
<span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="va">sam_names</span>, <span class="st">"_F_filt.fastq.gz"</span>, <span class="st">""</span><span class="op">)</span>
<span class="va">sam_names</span> <span class="op">&lt;-</span> <span class="fu">str_replace</span><span class="op">(</span><span class="va">sam_names</span>, <span class="st">"Connor-"</span>, <span class="st">"E"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">X</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">unique_names</span> <span class="op">&lt;-</span> <span class="va">sam_names</span>
<span class="va">aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"142"</span>,<span class="st">"143"</span>,<span class="st">"15"</span>,<span class="st">"ST"</span>,<span class="st">"22"</span>,<span class="st">"40"</span><span class="op">)</span>
<span class="va">non_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"33"</span>, <span class="st">"71"</span>, <span class="st">"106"</span><span class="op">)</span>

<span class="va">paired_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"E-142-1"</span>, <span class="st">"E142-1"</span>, <span class="st">"E-142-5"</span>, <span class="st">"E142-5"</span>, <span class="st">"E-142-10"</span>, <span class="st">"E142-10"</span>, <span class="st">"E-143-2"</span>, <span class="st">"E143-2"</span>, <span class="st">"E-143-7"</span>, <span class="st">"E143-7"</span>, <span class="st">"E-15-1"</span>, <span class="st">"E15-1"</span>, <span class="st">"ST-CAZ-4-R-O"</span>, <span class="st">"ST-CAZ-4-R-M"</span>, <span class="st">"ST-SAL-22-R-O"</span>, <span class="st">"ST-SAL-22-R-M"</span>, <span class="st">"ST-TRI-10-R-O"</span>, <span class="st">"ST-TRI-10-R-M"</span><span class="op">)</span>
<span class="va">paired_non_aster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"E33-7"</span>, <span class="st">"E-33-7"</span>, <span class="st">"E33-8"</span>, <span class="st">"E-33-8"</span>, <span class="st">"E33-9"</span>, <span class="st">"E-33-9"</span>, <span class="st">"E71-10"</span>, <span class="st">"E-71-10"</span>,<span class="st">"E71-2"</span>,<span class="st">"E-71-2"</span> ,<span class="st">"E71-3"</span>, <span class="st">"E-71-3"</span>, <span class="st">"E106-1"</span>, <span class="st">"E-106-1"</span>, <span class="st">"E106-3"</span>, <span class="st">"E-106-3"</span>, <span class="st">"E106-4"</span>, <span class="st">"E-106-4"</span><span class="op">)</span>
<span class="va">paired_specimens</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">paired_aster</span>, <span class="va">paired_non_aster</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We will use 9 colors for 9 different plants</span>
<span class="va">plant_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#999999"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span>, <span class="st">"#009E73"</span>, <span class="st">"#F0E442"</span>, <span class="st">"#0072B2"</span>, <span class="st">"#D55E00"</span>, <span class="st">"#CC79A7"</span>, <span class="st">"purple"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="ordination" class="section level2">
<h2 class="hasAnchor">
<a href="#ordination" class="anchor"></a>7.1 Ordination</h2>
<div id="multidimensional-scaling-mds-with-weighted-unifrac" class="section level3">
<h3 class="hasAnchor">
<a href="#multidimensional-scaling-mds-with-weighted-unifrac" class="anchor"></a>Multidimensional scaling (MDS) with weighted unifrac</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps_log</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/transformcounts.html">transform_sample_counts</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="va">x</span><span class="op">)</span>
  <span class="op">)</span>

<span class="va">out.wuf.log</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html">ordinate</a></span><span class="op">(</span>
  <span class="va">ps_log</span>, 
  method <span class="op">=</span> <span class="st">"MDS"</span>, 
  distance <span class="op">=</span> <span class="st">"wunifrac"</span>
  <span class="op">)</span>

<span class="va">evals</span> <span class="op">&lt;-</span> <span class="va">out.wuf.log</span><span class="op">$</span><span class="va">values</span><span class="op">$</span><span class="va">Eigenvalues</span>

<span class="va">mds_wuni</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_log</span>, 
  <span class="va">out.wuf.log</span>, 
  color <span class="op">=</span> <span class="st">"species_names"</span>, 
  shape <span class="op">=</span> <span class="st">"pna"</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>
    col <span class="op">=</span> <span class="st">"Plant type"</span>
    <span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">evals</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">/</span> <span class="va">evals</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_colour_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">Type</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>

<span class="co"># adding only paired-specimen names</span>
<span class="va">df_paired</span> <span class="op">&lt;-</span> <span class="va">mds_wuni</span><span class="op">$</span><span class="va">data</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
    <span class="va">unique_names</span> <span class="op">%in%</span> <span class="va">paired_specimens</span>
    <span class="op">)</span>

<span class="va">mds_wuni</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">df_paired</span>, 
    <span class="fu">aes</span><span class="op">(</span>
      x <span class="op">=</span> <span class="va">Axis.1</span>, 
      y <span class="op">=</span> <span class="va">Axis.2</span>, 
      label <span class="op">=</span> <span class="va">unique_names</span>
      <span class="op">)</span>, 
    color <span class="op">=</span> <span class="st">"black"</span>
    <span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-7-1.png" alt="Supplementary Figure 8: Multidimensional scaling (MDS) with weighted unifrac distance on all specimens." width="672"><p class="caption">
Supplementary Figure 8: Multidimensional scaling (MDS) with weighted unifrac distance on all specimens.
</p>
</div>
<p>We see that Centaurea solstitialis specimens are outlier that were sampled in a field across three countries, while other plants are grown from seeds. We label the paired specimens in all plants. The paired specimens are relatively close to each other, except (E-143-7, E143-7), (E-142-5, E142-5), (E-71-2, E71-2), and (E-71-3, E71-3) that are in positive and negative axes of the first two principal axes.</p>
<p>Fitzpatrick et al. (2018) concluded that there is no effect in within and between specimen bacterial diversity when they used universal or Aster-modified pPNA types.</p>
<p>The above Figure shows that specimens from the aster plants that are paired with universal or Aster-modified pPNA types make clusters.</p>
</div>
<div id="mds-with-bc" class="section level3">
<h3 class="hasAnchor">
<a href="#mds-with-bc" class="anchor"></a>MDS with BC</h3>
<p>An MDS plot using Bray-Curtis distance between specimens.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out.bc.log</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html">ordinate</a></span><span class="op">(</span>
  <span class="va">ps_log</span>, 
  method <span class="op">=</span> <span class="st">"MDS"</span>, 
  distance <span class="op">=</span> <span class="st">"bray"</span>
  <span class="op">)</span>

<span class="va">evals</span> <span class="op">&lt;-</span> <span class="va">out.bc.log</span><span class="op">$</span><span class="va">values</span><span class="op">$</span><span class="va">Eigenvalues</span>

<span class="va">mds_bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_log</span>, 
  <span class="va">out.bc.log</span>, 
  color <span class="op">=</span> <span class="st">"species_names"</span>, 
  shape <span class="op">=</span> <span class="st">"pna"</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">evals</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">/</span> <span class="va">evals</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>
    col <span class="op">=</span> <span class="st">"Plant type"</span>, 
    shape <span class="op">=</span> <span class="st">"pna"</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_colour_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">Type</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> 

<span class="va">df_paired</span> <span class="op">&lt;-</span> <span class="va">mds_bc</span><span class="op">$</span><span class="va">data</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
    <span class="va">unique_names</span> <span class="op">%in%</span> <span class="va">paired_specimens</span>
    <span class="op">)</span>

<span class="va">mds_bc</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">df_paired</span>, 
    <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Axis.1</span>, 
        y <span class="op">=</span> <span class="va">Axis.2</span>, 
        label <span class="op">=</span> <span class="va">unique_names</span>
        <span class="op">)</span>, 
    color <span class="op">=</span> <span class="st">"black"</span>
    <span class="op">)</span></code></pre></div>
<p><img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;"></p>
<p>The ordination of specimens in MDS with BC is similar to MDS with weighted unifrac.</p>
</div>
<div id="dpcoa" class="section level3">
<h3 class="hasAnchor">
<a href="#dpcoa" class="anchor"></a>DPCoA</h3>
<p>A DPCoA plot incorporates phylogenetic information.</p>
<p>The DPCoA specimens positions can be interpreted with respect to the ASVs coordinates in this display.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"ps_ans"</span><span class="op">)</span>
<span class="va">out.dpcoa.log</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html">ordinate</a></span><span class="op">(</span>
  <span class="va">ps_ans</span>, 
  method <span class="op">=</span> <span class="st">"DPCoA"</span>
  <span class="op">)</span>

<span class="va">evals</span> <span class="op">&lt;-</span> <span class="va">out.dpcoa.log</span><span class="op">$</span><span class="va">eig</span>

<span class="va">dpcoa_sam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_ans</span>,
  <span class="va">out.dpcoa.log</span>,
  color <span class="op">=</span> <span class="st">"species_names"</span>,
  shape <span class="op">=</span> <span class="st">"pna"</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">evals</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">/</span> <span class="va">evals</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>
    col <span class="op">=</span> <span class="st">"Plant type"</span>, 
    shape <span class="op">=</span> <span class="st">"pna"</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_colour_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">Type</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> 

<span class="va">df_paired</span> <span class="op">&lt;-</span> <span class="va">dpcoa_sam</span><span class="op">$</span><span class="va">data</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
    <span class="va">unique_names</span> <span class="op">%in%</span> <span class="va">paired_specimens</span>
    <span class="op">)</span>

<span class="va">dpcoa_sam</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">df_paired</span>,
    <span class="fu">aes</span><span class="op">(</span>
      x <span class="op">=</span> <span class="va">CS1</span>, 
      y <span class="op">=</span> <span class="va">CS2</span>,
      label <span class="op">=</span> <span class="va">unique_names</span>
      <span class="op">)</span>, 
    color <span class="op">=</span> <span class="st">"black"</span>
    <span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-9-1.png" alt="Supplementary Figure 9: (A) A DPCoA plot that incorporates phylogenetic information." width="672"><p class="caption">
Supplementary Figure 9: (A) A DPCoA plot that incorporates phylogenetic information.
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dpcoa_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_ans</span>, 
  <span class="va">out.dpcoa.log</span>,
  type <span class="op">=</span> <span class="st">"species"</span>, 
  color <span class="op">=</span> <span class="st">"Class"</span>
  <span class="op">)</span> 
  

<span class="va">df_taxa</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
  <span class="va">dpcoa_taxa</span><span class="op">$</span><span class="va">data</span>, 
  <span class="op">(</span><span class="va">CS1</span> <span class="op">&gt;</span> <span class="fl">0.63</span> <span class="op">|</span> <span class="va">CS1</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">0.14</span> <span class="op">|</span> <span class="va">CS2</span> <span class="op">&gt;</span> <span class="fl">0.29</span> <span class="op">|</span> <span class="va">CS2</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">0.13</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span>
    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">Class</span><span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span>

<span class="va">dpcoa_taxa</span><span class="op">$</span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">df_taxa</span>

<span class="va">dpcoa_taxa</span> <span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span><span class="va">evals</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">/</span> <span class="va">evals</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/randomcoloR/man/distinctColorPalette.html">distinctColorPalette</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span>
        <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span>
          <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_ans</span><span class="op">)</span><span class="op">[</span>, <span class="st">"Class"</span><span class="op">]</span>
          <span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-9-2.png" alt="Supplementary Figure 9: (A) A DPCoA plot that incorporates phylogenetic information." width="672"><p class="caption">
Supplementary Figure 9: (A) A DPCoA plot that incorporates phylogenetic information.
</p>
</div>
</div>
</div>
<div id="integrating-the-phylogenetic-tree-into-the-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#integrating-the-phylogenetic-tree-into-the-analyses" class="anchor"></a>7.2 Integrating the phylogenetic tree into the analyses</h2>
<div id="adaptive-gpca" class="section level3">
<h3 class="hasAnchor">
<a href="#adaptive-gpca" class="anchor"></a>Adaptive gPCA</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/adaptiveGPCA/man/processPhyloseq.html">processPhyloseq</a></span><span class="op">(</span>
  <span class="va">ps</span>
  <span class="op">)</span>
<span class="va">out.agpca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/adaptiveGPCA/man/adaptivegpca.html">adaptivegpca</a></span><span class="op">(</span>
  <span class="va">pp</span><span class="op">$</span><span class="va">X</span>, 
  <span class="va">pp</span><span class="op">$</span><span class="va">Q</span>, 
  k <span class="op">=</span> <span class="fl">2</span>
  <span class="op">)</span>

<span class="va">agPCA_samples</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    <span class="va">out.agpca</span><span class="op">$</span><span class="va">U</span>, 
    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_point</span><span class="op">(</span>
      <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Axis1</span>,
          y <span class="op">=</span> <span class="va">Axis2</span>, 
          color <span class="op">=</span> <span class="va">species_names</span>,
          shape <span class="op">=</span> <span class="va">pna</span><span class="op">)</span>,
      size <span class="op">=</span> <span class="fl">3</span>
      <span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>
    x <span class="op">=</span> <span class="st">"Axis 1"</span>, 
    y <span class="op">=</span> <span class="st">"Axis 2"</span>,
    col <span class="op">=</span> <span class="st">"Plant type"</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_colour_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span> <span class="op">+</span>
  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span> <span class="va">Type</span><span class="op">)</span>

<span class="va">agPCA_samples</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-10-1.png" alt="Figure 4: The results from adaptive gPCA reveal Centaurea solstitialis specimens are outliers among Asteraceae plants in the left of Axis 1. Axis 2 explains the microbial variability in plant types." width="672"><p class="caption">
Figure 4: The results from adaptive gPCA reveal Centaurea solstitialis specimens are outliers among Asteraceae plants in the left of Axis 1. Axis 2 explains the microbial variability in plant types.
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">agPCA_asv</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
    <span class="va">out.agpca</span><span class="op">$</span><span class="va">QV</span>, 
    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>
    <span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_point</span><span class="op">(</span>
      <span class="fu">aes</span><span class="op">(</span>
        x <span class="op">=</span> <span class="va">Axis1</span>, 
        y <span class="op">=</span> <span class="va">Axis2</span>, 
        color <span class="op">=</span> <span class="va">Class</span>
        <span class="op">)</span>, 
      size <span class="op">=</span> <span class="fl">3</span>
      <span class="op">)</span> <span class="op">+</span>
    <span class="fu">xlab</span><span class="op">(</span>
      <span class="st">"Axis 1"</span>
      <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ylab</span><span class="op">(</span>
    <span class="st">"Axis 2"</span>
    <span class="op">)</span>

<span class="va">main_class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bacilli"</span>, 
                <span class="st">"Actinobacteria"</span>, 
                <span class="st">"Cytophagia"</span>, 
                <span class="st">"Gammaproteobacteria"</span>, 
                <span class="st">"Betaproteobacteria"</span>, 
                <span class="st">"Deltaproteobacteria"</span>, 
                <span class="st">"Alphaproteobacteria"</span>, 
                <span class="st">"Opitutae"</span><span class="op">)</span>

<span class="va">df_taxa</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
  <span class="va">agPCA_asv</span><span class="op">$</span><span class="va">data</span>, 
  <span class="va">Class</span> <span class="op">%in%</span> <span class="va">main_class</span>
  <span class="op">)</span>

<span class="va">agPCA_asv</span><span class="op">$</span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">df_taxa</span>

<span class="va">agPCA_asv</span> <span class="op">&lt;-</span> <span class="va">agPCA_asv</span> <span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/randomcoloR/man/distinctColorPalette.html">distinctColorPalette</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span>
        <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span>
          <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">ps_ans</span><span class="op">)</span><span class="op">[</span>, <span class="st">"Class"</span><span class="op">]</span><span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span>
<span class="va">agPCA_asv</span></code></pre></div>
<p><img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;"></p>
<p><em>Centaurea solstitialis</em> specimens are outliers among Asteraceae plants on the left of Axis 1. Axis 2 explains the microbial variability in plant types.</p>
</div>
</div>
<div id="correspondence-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#correspondence-analysis" class="anchor"></a>7.3 Correspondence analysis</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps_ca</span> <span class="op">&lt;-</span> <span class="va">ps</span>
<span class="va">out.ca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/ordinate.html">ordinate</a></span><span class="op">(</span>
  <span class="va">ps_ca</span>, 
  method <span class="op">=</span> <span class="st">"CCA"</span>
  <span class="op">)</span>

<span class="va">evals</span> <span class="op">&lt;-</span> <span class="va">out.ca</span><span class="op">$</span><span class="va">CA</span><span class="op">$</span><span class="va">eig</span>
<span class="va">evals_prop</span> <span class="op">&lt;-</span> <span class="va">evals</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">evals</span><span class="op">)</span>

<span class="va">ca_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_ca</span>,
  <span class="va">out.ca</span>,
  type <span class="op">=</span> <span class="st">"sample"</span>,
  color <span class="op">=</span> <span class="st">"species_names"</span>,
  shape <span class="op">=</span> <span class="st">"pna"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>
    col <span class="op">=</span> <span class="st">"Plant type"</span>,
    shape <span class="op">=</span> <span class="st">"pna"</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">facet_grid</span><span class="op">(</span><span class="op">~</span><span class="va">Type</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span><span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">theme_update</span><span class="op">(</span>
    text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,
    panel.spacing <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">2</span>, <span class="st">"lines"</span><span class="op">)</span>
<span class="op">)</span>


<span class="va">ca_plot</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-12-1.png" alt="Figure 5: Correspondence analysis of all plant specimens. (A)" width="672"><p class="caption">
Figure 5: Correspondence analysis of all plant specimens. (A)
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ca_plot_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_ordination.html">plot_ordination</a></span><span class="op">(</span>
  <span class="va">ps_ca</span>,
  <span class="va">out.ca</span>,
  type <span class="op">=</span> <span class="st">"species"</span>,
  color <span class="op">=</span> <span class="st">"Class"</span>
  <span class="op">)</span> 

<span class="va">main_class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bacilli"</span>, 
                <span class="st">"Actinobacteria"</span>, 
                <span class="st">"Cytophagia"</span>, 
                <span class="st">"Gammaproteobacteria"</span>,
                <span class="st">"Betaproteobacteria"</span>,
                <span class="st">"Deltaproteobacteria"</span>,
                <span class="st">"Alphaproteobacteria"</span>, 
                <span class="st">"Opitutae"</span>, 
                <span class="st">"Sphingobacteriia"</span><span class="op">)</span>

<span class="va">df_taxa</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
  <span class="va">ca_plot_taxa</span><span class="op">$</span><span class="va">data</span>, 
  <span class="va">Class</span> <span class="op">%in%</span> <span class="va">main_class</span>
  <span class="op">)</span>

<span class="va">df_taxa</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">df_taxa</span><span class="op">$</span><span class="va">Class</span><span class="op">)</span>

<span class="va">ca_plot_taxa</span><span class="op">$</span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">df_taxa</span>

<span class="va">ca_plot_taxa</span> <span class="op">&lt;-</span> <span class="va">ca_plot_taxa</span> <span class="op">+</span>
  <span class="fu">coord_fixed</span><span class="op">(</span><span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"Class"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_point</span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_jitter</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/randomcoloR/man/distinctColorPalette.html">distinctColorPalette</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span>
        <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span>
          <span class="va">main_class</span>
          <span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">)</span>

<span class="va">ca_plot_taxa</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-13-1.png" alt="Figure 5: Correspondence analysis of all plant specimens. (B)" width="672"><p class="caption">
Figure 5: Correspondence analysis of all plant specimens. (B)
</p>
</div>
</div>
<div id="network-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#network-analysis" class="anchor"></a>7.4 Network analysis</h2>
<div id="network-analysis-on-paired-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#network-analysis-on-paired-specimens" class="anchor"></a>Network analysis on paired specimens</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_samples-methods.html">subset_samples</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  <span class="va">unique_names</span> <span class="op">%in%</span> <span class="va">paired_specimens</span>
  <span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html">prune_taxa</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_sums.html">taxa_sums</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>, 
  <span class="va">ps</span>
  <span class="op">)</span>
<span class="va">ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 1348 taxa and 36 samples ]
## sample_data() Sample Data:       [ 36 samples by 16 sample variables ]
## tax_table()   Taxonomy Table:    [ 1348 taxa by 6 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 1348 tips and 1347 internal nodes ]</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/make_network.html">make_network</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  max.dist <span class="op">=</span> <span class="fl">0.8</span>
  <span class="op">)</span>

<span class="va">sampledata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">$</span><span class="va">unique_names</span> <span class="op">&lt;-</span> <span class="va">sampledata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">)</span>, <span class="st">"unique_names"</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">$</span><span class="va">pna</span> <span class="op">&lt;-</span> <span class="va">sampledata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">)</span>, <span class="st">"pna"</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">$</span><span class="va">species_names</span> <span class="op">&lt;-</span> <span class="va">sampledata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">net</span><span class="op">)</span><span class="op">)</span>, <span class="st">"species_names"</span><span class="op">]</span>

<span class="va">p_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/plot_network.html">plot_network</a></span><span class="op">(</span>
  <span class="va">net</span>, 
  <span class="va">ps</span>, 
  color <span class="op">=</span> <span class="st">"species_names"</span>,
  shape <span class="op">=</span> <span class="st">"pna"</span>
  <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">labs</span><span class="op">(</span>
    col <span class="op">=</span> <span class="st">"Species names"</span>
    <span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_colour_manual</span><span class="op">(</span>
    values <span class="op">=</span> <span class="va">plant_colors</span>
    <span class="op">)</span>
<span class="va">p_net</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="07_multivariate_and_network_analysis_files/figure-html/unnamed-chunk-15-1.png" alt="Supplementary Figure 10: A network created by thresholding the Jaccard dissimilarity matrix at 0.8." width="672"><p class="caption">
Supplementary Figure 10: A network created by thresholding the Jaccard dissimilarity matrix at 0.8.
</p>
</div>
</div>
<div id="graph-based-two-sample-tests" class="section level3">
<h3 class="hasAnchor">
<a href="#graph-based-two-sample-tests" class="anchor"></a>Graph-based two-sample tests</h3>
<p>We first perform a test using an Minimum spanning tree (MST) with Jaccard dissimilarity. We want to know whether the universal or Asteraceae-modified pPNAs come from the same distribution in paired-specimens.</p>
<p>Since there is a grouping in the data by specimen, we can’t simply permute all the labels, we need to maintain this nested structure – this is what the grouping argument does. Here we permute the pna labels but keep the plant type structure intact.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">15</span><span class="op">)</span>
<span class="va">gt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseqGraphTest/man/graph_perm_test.html">graph_perm_test</a></span><span class="op">(</span>
  <span class="va">ps</span>, 
  sampletype <span class="op">=</span> <span class="st">"pna"</span>, 
  grouping <span class="op">=</span> <span class="st">"unique_names"</span>,
  distance <span class="op">=</span> <span class="st">"jaccard"</span>, 
  type <span class="op">=</span> <span class="st">"mst"</span><span class="op">)</span>
<span class="va">gt</span><span class="op">$</span><span class="va">pval</span></code></pre></div>
<pre><code>## [1] 0.882</code></pre>
<p>This test has a larger p-value, and we do not reject the null hypothesis that the paired-specimens come from the same distribution.</p>
<p>This result mirrors the observations from the PCA after the truncated-rank transformation, adaptive gPCA, MDS, and DPCoA that paired-specimens have similar microbial variability.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pratheepa Jeganathan, Susan Holmes.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
