<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analzying molecular microbiome data with diffTop • diffTop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analzying molecular microbiome data with diffTop">
<meta property="og:description" content="diffTop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">diffTop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/diffTop.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/appendix/01_introduction.html">Section 1: INTRODUCTION</a>
    </li>
    <li>
      <a href="../articles/appendix/02_microbiome_data.html">Section 2: MICROBIOME DATA</a>
    </li>
    <li>
      <a href="../articles/appendix/03_example_16S_data.html">Section 3: EXAMPLE OF 16S rRNA GENE SEQUENCING DATA</a>
    </li>
    <li>
      <a href="../articles/appendix/04_models.html">Section 4: MODELS</a>
    </li>
    <li>
      <a href="../articles/appendix/05_transformation.html">Section 5: TRANSFORMATIONS</a>
    </li>
    <li>
      <a href="../articles/appendix/06_visualization.html">Section 6: VISUALIZATIONS FOR HETEROGENEOUS DATA</a>
    </li>
    <li>
      <a href="../articles/appendix/07_multivariate_and_network_analysis.html">Section 7: MULTIVARIATE AND NETWORK ANALYSES</a>
    </li>
    <li>
      <a href="../articles/appendix/08_differential_abundance_analysis.html">Section 8: DIFFERENTIAL ABUNDANCE ANALYSIS</a>
    </li>
    <li>
      <a href="../articles/appendix/09_latent_Dirichlet_allocation.html">Section 9: LATENT DIRICHLET ALLOCATION</a>
    </li>
    <li>
      <a href="../articles/appendix/10_uncertainity_quan_ordination.html">Section 10: UNCERTAINTY QUANTIFICATION FOR ORDINATION ANALYSIS</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="diffTop_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analzying molecular microbiome data with diffTop</h1>
                        <h4 class="author">Pratheepa Jeganathan and Susan Holmes</h4>
            
            <h4 class="date">08/04/2021</h4>
      
      
      <div class="hidden name"><code>diffTop.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>One of the important tasks in molecular microbiome data is the quantification and statistical inference of micriobial communities across envrionment or groups. The count data are presented as a contingency table, for each sample the number of sequence fragments that have been assigned to each taxon. The package diffTop provides methods to estimate topics using latent Dirichlet allocation, align posterior samples across chains, find topic differences across environments by use of generalized linear models, visualize the topic and taxa proportions. This vignette explains the use of the package and demonstrates typical workflows.</p>
    </div>
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">diffTop</span><span class="op">)</span></code></pre></div>
<div id="standard-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#standard-workflow" class="anchor"></a>Standard workflow</h1>
<p><strong>Note:</strong> if you use diffTop in published research, please cite:</p>
<blockquote>
<p>Jeganathan, P. and Holmes, S.P. (2021) A statistical perspective on the challenges in molecular microbial biology. <em>Journal of Agricultural, Biological and Environmental Statistics</em>, <strong>26(2)</strong>:131 - 160. <a href="https://doi.org/10.1007/s13253-021-00447-1">10.1007/s13253-021-00447-1</a></p>
</blockquote>
<div id="quick-start" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick start</h2>
<p>Here we show the steps for the differential topic analysis. The following code assume that the <code>phyloseq</code> object is available. Please refer to <a href="https://www.bioconductor.org/packages/release/bioc/html/phyloseq.html"><code>phyloseq</code></a> for creating a <code>phyloseq</code> class object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K</span> <span class="op">&lt;-</span> <span class="fl">11</span>
<span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">gamma</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">psE_BARBI</span>
<span class="va">stan.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setUpData.html">setUpData</a></span><span class="op">(</span>
  ps <span class="op">=</span> <span class="va">ps</span>,
  K <span class="op">=</span> <span class="va">K</span>,
  alpha <span class="op">=</span> <span class="va">alpha</span>,
  gamma <span class="op">=</span> <span class="va">gamma</span>
<span class="op">)</span>

<span class="va">iter</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="va">chains</span> <span class="op">&lt;-</span> <span class="fl">4</span>
<span class="va">stan.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LDAtopicmodel.html">LDAtopicmodel</a></span><span class="op">(</span>
  stan_data <span class="op">=</span> <span class="va">stan.data</span>, 
  iter <span class="op">=</span> <span class="va">iter</span>, 
  chains <span class="op">=</span> <span class="va">chains</span>, 
  sample_file <span class="op">=</span> <span class="cn">NULL</span>,
  diagnostic_file <span class="op">=</span> <span class="cn">NULL</span>,
  cores <span class="op">=</span> <span class="fl">4</span>,
  control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>adapt_delta <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>,
  save_dso <span class="op">=</span> <span class="cn">TRUE</span>,
  algorithm <span class="op">=</span> <span class="st">"NUTS"</span>
  <span class="op">)</span>

<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">rstan</span><span class="fu">::</span><span class="fu"><a href="https://mc-stan.org/rstan/reference/stanfit-method-extract.html">extract</a></span><span class="op">(</span>
  <span class="va">stan.fit</span>, 
  permuted <span class="op">=</span> <span class="cn">TRUE</span>, 
  inc_warmup <span class="op">=</span> <span class="cn">FALSE</span>, 
  include <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span>

<span class="va">theta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">theta</span> 

<span class="va">aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentMatrix.html">alignmentMatrix</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">ps</span>, 
  <span class="va">K</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  chain <span class="op">=</span> <span class="va">chains</span>,
  SampleID_name <span class="op">=</span> <span class="st">"unique_names"</span>
  <span class="op">)</span>

<span class="va">theta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/thetaAligned.html">thetaAligned</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span>

<span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">beta</span> 

<span class="va">beta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/betaAligned.html">betaAligned</a></span><span class="op">(</span>
  <span class="va">beta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span> 


<span class="va">dds_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">psE_BARBI</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">res_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all</span><span class="op">)</span>
<span class="va">res_all</span></code></pre></div>
</div>
<div id="input-data" class="section level2">
<h2 class="hasAnchor">
<a href="#input-data" class="anchor"></a>Input data</h2>
<p>We set-up data for latent Dirichlet allocation (LDA) based on a <code>phyloseq</code> object, the number of topics, hyperparameters for topic proportion in each sample and ASVs proportion in each topic.</p>
<p>We set the hyperparameters <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\gamma\)</span> less than one to generate mixtures that are different from each other. We choose 0.8 for <span class="math inline">\(\alpha\)</span> across all samples so that we avoid generating unrealistic topics.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"psE_BARBI"</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">psE_BARBI</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K</span> <span class="op">&lt;-</span> <span class="fl">11</span>
<span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">gamma</span> <span class="op">&lt;-</span> <span class="fl">0.8</span>
<span class="va">stan.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setUpData.html">setUpData</a></span><span class="op">(</span>
  ps <span class="op">=</span> <span class="va">ps</span>,
  K <span class="op">=</span> <span class="va">K</span>,
  alpha <span class="op">=</span> <span class="va">alpha</span>,
  gamma <span class="op">=</span> <span class="va">gamma</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="posterior-sampling" class="section level2">
<h2 class="hasAnchor">
<a href="#posterior-sampling" class="anchor"></a>Posterior sampling</h2>
<p>We need to estimate <span class="math inline">\(\theta\)</span> and <span class="math inline">\(B = \left[\beta_{1}, \cdots, \beta_{K} \right]^{T}.\)</span></p>
<p>We estimate the parameters using HMC NUTS with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warm-up samples.</p>
<p><code>LDAtopicmodel</code> uses an LDA model written in lda.stan. The user doesn’t need to write the Stan model. We use <code>stan</code> function from <code>rstan</code> package. Please refer to <code><a href="https://mc-stan.org/rstan/reference/stan.html">?stan</a></code> to specify the inputs for the <code>stan</code> function. For the whole dataset, we can run the following chunk in high-performance computing and save the results.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">iter</span> <span class="op">&lt;-</span> <span class="fl">2000</span>
<span class="va">chains</span> <span class="op">&lt;-</span> <span class="fl">4</span>

<span class="va">stan.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LDAtopicmodel.html">LDAtopicmodel</a></span><span class="op">(</span>
  stan_data <span class="op">=</span> <span class="va">stan.data</span>, 
  iter <span class="op">=</span> <span class="va">iter</span>, 
  chains <span class="op">=</span> <span class="va">chains</span>, 
  sample_file <span class="op">=</span> <span class="cn">NULL</span>,
  diagnostic_file <span class="op">=</span> <span class="cn">NULL</span>,
  cores <span class="op">=</span> <span class="fl">4</span>,
  control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>adapt_delta <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>,
  save_dso <span class="op">=</span> <span class="cn">TRUE</span>,
  algorithm <span class="op">=</span> <span class="st">"NUTS"</span>
  <span class="op">)</span></code></pre></div>
<p>We extract posterior samples from the results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">rstan</span><span class="fu">::</span><span class="fu"><a href="https://mc-stan.org/rstan/reference/stanfit-method-extract.html">extract</a></span><span class="op">(</span>
  <span class="va">stan.fit</span>, 
  permuted <span class="op">=</span> <span class="cn">TRUE</span>, 
  inc_warmup <span class="op">=</span> <span class="cn">FALSE</span>, 
  include <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span></code></pre></div>
<div id="alignment" class="section level3">
<h3 class="hasAnchor">
<a href="#alignment" class="anchor"></a>Alignment</h3>
<p>We estimated the parameters using the HMC-NUTS sampler with four chains and 2000 iterations. Out of these 2000 iterations, 1000 iterations were used as warm-up samples and discarded. Label switching across chains makes it difficult to directly compute log predictive density, split-<span class="math inline">\(\hat{R}\)</span>, effective sample size for model assessment, and evaluate convergence and mixing of chains.</p>
<p>To address this issue, we fixed the order of topics in chain one and then found the permutation that best aligned the topics across all four chains. For each chain two to four, we identified the estimated topics pair with the highest correlation, then found the next highest pair among the remaining, and so forth.</p>
<p>We create a Topic <span class="math inline">\(*\)</span> Chain matrix.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">theta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">theta</span> 

<span class="va">aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentMatrix.html">alignmentMatrix</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">ps</span>, 
  <span class="va">K</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  chain <span class="op">=</span> <span class="va">chains</span>,
  SampleID_name <span class="op">=</span> <span class="st">"unique_names"</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="align-topic-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#align-topic-proportion" class="anchor"></a>Align topic proportion</h3>
<p>We align the topic proportion in each specimen across four chains.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">theta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/thetaAligned.html">thetaAligned</a></span><span class="op">(</span>
  <span class="va">theta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="align-asv-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#align-asv-proportion" class="anchor"></a>Align ASV proportion</h3>
<p>We align the ASVs proportion in each topic across four chains.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># an array (iterations *topic * ASV)</span>
<span class="va">beta</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">beta</span> 

<span class="co"># an array (iterations *topic * ASV)</span>
<span class="va">beta_aligned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/betaAligned.html">betaAligned</a></span><span class="op">(</span>
  <span class="va">beta</span>, 
  <span class="va">K</span>, 
  <span class="va">aligned</span>, 
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>, 
  chain <span class="op">=</span> <span class="va">chains</span>
  <span class="op">)</span> </code></pre></div>
</div>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div id="plot-topic-proportion" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-topic-proportion" class="anchor"></a>Plot topic proportion</h3>
<p>Plot the topic proportion in specimens and we can save it for publication purposes. We can draw an informative summary of bacterial communities in each phenotype.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotTopicProportion.html">plotTopicProportion</a></span><span class="op">(</span>
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  <span class="va">K</span>,
  col_names_theta_all <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"iteration"</span>, <span class="st">"Sample"</span>, <span class="st">"Topic"</span>, <span class="st">"topic.dis"</span><span class="op">)</span>,
  chain <span class="op">=</span> <span class="fl">4</span>,
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  SampleIDVarInphyloseq <span class="op">=</span> <span class="st">"unique_names"</span>,
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="plot-asvs-distribution" class="section level3">
<h3 class="hasAnchor">
<a href="#plot-asvs-distribution" class="anchor"></a>Plot ASVs distribution</h3>
<p>We plot the ASVs distribution in each topic.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plotASVCircular.html">plotASVCircular</a></span><span class="op">(</span>
  <span class="va">ps</span>,
  <span class="va">K</span>,
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  chain <span class="op">=</span> <span class="va">chains</span>,
  <span class="va">beta_aligned</span>,
  varnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"iterations"</span>, <span class="st">"topic"</span>, <span class="st">"rsv_ix"</span><span class="op">)</span>,
  value_name <span class="op">=</span> <span class="st">"beta_h"</span>,
  taxonomylevel <span class="op">=</span> <span class="st">"Class"</span>,
  thresholdASVprop <span class="op">=</span> <span class="fl">0.008</span>
  <span class="op">)</span></code></pre></div>
</div>
</div>
<div id="model-assessement" class="section level2">
<h2 class="hasAnchor">
<a href="#model-assessement" class="anchor"></a>Model assessement</h2>
<p>We do goodness of fit test using the posterior estimates and observed data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/modelAssessment.html">modelAssessment</a></span><span class="op">(</span>
  <span class="va">ps</span>,
  <span class="va">stan.fit</span>,
  iterUse <span class="op">=</span> <span class="va">iter</span><span class="op">/</span><span class="fl">2</span>,
  statistic <span class="op">=</span> <span class="st">"max"</span>,
  ASVsIndexToPlot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">10</span><span class="op">:</span><span class="fl">14</span>, <span class="fl">19</span><span class="op">:</span><span class="fl">26</span>, <span class="fl">36</span>, <span class="fl">51</span><span class="op">:</span><span class="fl">53</span>, <span class="fl">148</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="diagnostics" class="section level2">
<h2 class="hasAnchor">
<a href="#diagnostics" class="anchor"></a>Diagnostics</h2>
<div id="effective-sample-size-ess" class="section level3">
<h3 class="hasAnchor">
<a href="#effective-sample-size-ess" class="anchor"></a>Effective sample size (ESS)</h3>
<p>We compute effective sample size for each parameter.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">diagnosticsPlot</span>(</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">theta_aligned =</span> theta_aligned,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">beta_aligned =</span> beta_aligned</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">iterUse =</span> iter<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chain =</span> chains</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>p_ess_bulk <span class="ot">&lt;-</span> p[[[<span class="dv">1</span>]]]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>p_ess_bulk</span></code></pre></div>
</div>
<div id="hatr" class="section level3">
<h3 class="hasAnchor">
<a href="#hatr" class="anchor"></a><span class="math inline">\(\hat{R}\)</span>
</h3>
<p>We can check whether <span class="math inline">\(\hat{R}\)</span> is less than 1.02</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p_rhat</span> <span class="op">&lt;-</span> <span class="va">p</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>
<span class="va">p_rhat</span></code></pre></div>
</div>
</div>
<div id="differential-topic-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-topic-analysis" class="anchor"></a>Differential topic analysis</h2>
<div id="test-on-all-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-all-specimens" class="anchor"></a>Test on all specimens</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>,
  fitType <span class="op">=</span> <span class="st">"mean"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all</span><span class="op">)</span></code></pre></div>
</div>
<div id="test-on-selected-specimens" class="section level3">
<h3 class="hasAnchor">
<a href="#test-on-selected-specimens" class="anchor"></a>Test on selected specimens</h3>
<p>Let’s test on the paired-specimens.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dds_all_paired</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffTopAnalysis.html">diffTopAnalysis</a></span><span class="op">(</span>
  design <span class="op">=</span> <span class="op">~</span> <span class="va">pna</span>,
  <span class="va">ps</span>,
  <span class="va">theta_aligned</span>,
  subsetSample <span class="op">=</span> <span class="va">paired_specimens</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds_all_paired</span><span class="op">)</span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pratheepa Jeganathan, Susan Holmes.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
